### PR TITLE
Sketch a Span.log_kv method

### PR DESCRIPTION
**Pull request by bensigelman**
Per opentracing/opentracing.github.io#96

This is only a sketch... once we decide on a specific API I will adjust tests and so on.

### CODE REVIEW

File: opentracing/span.py [hunk 1]
```diff
@@ -203,3 +205,11 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                                                 'val': exc_val,
                                                 'tb': exc_tb})
         self.finish()
+
+    def log_event(self, event, payload=None):
+        """DEPRECATED"""
+        return self
```
Comment (yurishkuro) [inline opentracing/span.py]: even if deprecated, we can provide default behavior right here by delegating back to `log_kv`

File: opentracing/span.py [hunk 2]
```diff
@@ -203,3 +205,11 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+
+    def log(self, **kwargs):
```
Comment (yurishkuro) [inline opentracing/span.py]: kind of a shame to kill this one. It's nearly equivalent to `log_kv` except requires "safe" field labels, which incidentally is all you have in the examples for `log_kv`

File: opentracing/span.py [hunk 3]
```diff
@@ -129,24 +129,26 @@ def set_tag(self, key, value):
...
-        :return: returns the span itself, for chaining the calls
-        """
-        return self
+        For example,
 
-    def log(self, **kwargs):
-        """Records a generic Log event at an arbitrary timestamp.
+            span.log_kv({
+                "event": "time to first byte",
+                "packet_size": packet.size()})
```
Comment (yurishkuro) [inline opentracing/span.py]: For illustration purposes I'd suggest using labels that cannot be simply passed as `**kwargs`, like `packet.size`

**Comment by yurishkuro** [comment]
👍

File: opentracing/span.py (hunk 3 above)
Comment (bensigelman) [inline opentracing/span.py]: good idea

File: opentracing/span.py (hunk 2 above)
Comment (bensigelman) [inline opentracing/span.py]: Yeah, I just delegated the deprecated methods in the latest commit.

**Comment by bensigelman** [comment]
@yurishkuro I updated the tests and harness to accommodate log_kv... PTAL.

I am also about to bump the API minor version (but not major since this is just the deprecation).

**Comment by yurishkuro** [comment]
:shipit:

**Pull request merged**